files/patch.py

```diff
@@ -98,10 +104,12 @@ class PatchError(Exception):
     pass
 
 
-def is_already_applied(patch_func, patch_file, basedir, dest_file=None, strip=0):
+def is_already_applied(patch_func, patch_file, basedir, dest_file=None, binary=False, strip=0):
     opts = ['--quiet', '--reverse', '--forward', '--dry-run',
             "--strip=%s" % strip, "--directory='%s'" % basedir,
             "--input='%s'" % patch_file]
+    if binary:
+        opts.append("--binary")
```

Comment:
Please use single quotes around string without formatting, to be consistent with code-style used in this file.